django_freeradius/api/serializers.py

```diff
@@ -90,7 +90,16 @@ class RadiusBatchSerializer(serializers.ModelSerializer):
     pdf = serializers.FileField(required=False, read_only=True)
     number_of_users = serializers.IntegerField(required=False,
                                                write_only=True,
-                                               min_value=1)
+                                               min_value=1,)
+
+    def validate(self, data):
+
+        if data['strategy'] == 'prefix' and data.get('number_of_users', '') == '':
+            raise serializers.ValidationError(
+                {'number_of_users': 'The field number_of_users cannot be empty'}
+            )
+        else:
+            return super(RadiusBatchSerializer, self).validate(data)
```

Comment:
can you try replacing this with:

```python
def validate(self, data):
    if data['strategy'] == 'prefix' and not data.get('number_of_users'):
        raise serializers.ValidationError({
            'number_of_users': 'The field number_of_users cannot be empty'
        })
    return super().validate(data)
```

we can use the form `super()` because this module is python3 only.

Reply:
I think `data.get('number_of_users')` should be replaced with `data.get('number_of_users`, '')` because I noticed that if the api request dosen't have the `number_of_users` attribute passed in then it's not available as a key in `data` @nemesisdesign .

Reply:
then it would return `None` which would be falsy so that should be ok